### DISCUSSION TITLE
Invalid hook call. Hooks can only be called inside of the body of a function component

### DISCUSSION THREAD

**Discussion by naga-wb** [vercel/next.js#17199] [Help] [answered]
Hi,

I created a next app inside a react app folder. Because the plan to migrate next app and also move frontend code separately(currently all Rails and FE are in same directory). 

I moved few react  components to next app. I fixed all dependency issues and run the application. I get invalid hook error when I access component  which is located inside next app folder from root directory.

FYI... I used same version of react and reactDOM in both root and next folders. 

How can I resolve this issue?

"next": "9.5.3",
 "react": "16.13.1",
 "react-dom": "16.13.1",

**Comment by jamesmosier** [comment, marked as answer]
By including 2 versions of React you will see this bug. There is an issue open though which sounds similar https://github.com/vercel/next.js/issues/9022 as well as https://github.com/vercel/next.js/issues/11363

Aside from that I found [quite a few other issues that might help](https://github.com/vercel/next.js/issues?q=is%3Aissue+Hooks+can+only+be+called+inside+of+the+body+)
